docs/book/src/developer/architecture/controllers/support-multiple-instances.md

```diff
@@ -0,0 +1,33 @@
+# Support running multiple instances of the same provider
+
+In releases <= v1alpha3 the need of supporting multiple credentials was addressed by running multiple
+instances of the same provider, each one with its own credentials.
+
+However, running multiple instances of the same provider proved to be complicated for several reason like e.g.
+
+- Complexity in packaging providers, due to the requirement to split CRDs and the web-hooks from the controller given that
+  only the controller can actually be installed many times, while CRDs and web-hooks can exist only in one instance.
+- Complexity in deploying providers, due to the requirement to ensure consistency of the management cluster e.g. avoid
+  controllers watching for the same namespaces, etc. 
+- The introduction of the concept of management groups in clusterctl, with impacts on the user experience/documentation.
+- Complexity in managing co-existence of different versions of the same provider while there could be only
+  one version of CRDs and web-hooks. 
+- A residual risk while running different version of the same provider, because some version of the provider
+  de-facto were forced to run with CRDs and web-hooks deployed from a different version.
+
+Nevertheless, we want to make it possible for users to choose to deploy multiple instances of the same providers, 
+in case the above limitations/extra complexity are acceptable for them.
+
+## Contract
+
+In order to make it possible for users to deploy multiple instances of the same provider:
+
+- All the provider will be required to support the `--namespace` flag in their controllers.
+
+Users selecting this deployment model should be aware that:
+
+- In the Cluster API project, we are not going to include release a `component.yaml` file supporting this scenario; 
+  however it will be possible to create such file starting from the content of the `/config` folder.
+- clusterctl and the CAPI provider operator are not going to support deploying/managing multiple instances of the same provider.
```

Comment:
```suggestion
```

Reply:
I think we should state this in order to set clear expectations, unless you think this is clear from the previous statement